pkg/components/bindings/output_pluggable.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2022 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bindings
+
+import (
+	"context"
+
+	"github.com/dapr/components-contrib/bindings"
+	"github.com/dapr/dapr/pkg/components"
+	"github.com/dapr/dapr/pkg/components/pluggable"
+	proto "github.com/dapr/dapr/pkg/proto/components/v1"
+	"github.com/dapr/kit/logger"
+)
+
+// grpcOutputBinding is a implementation of a outputbinding over a gRPC Protocol.
+type grpcOutputBinding struct {
+	*pluggable.GRPCConnector[proto.OutputBindingClient]
+	bindings.OutputBinding
+	operations []bindings.OperationKind
+}
+
+// Init initializes the grpc outputbinding passing out the metadata to the grpc component.
+func (b *grpcOutputBinding) Init(metadata bindings.Metadata) error {
+	if err := b.Dial(metadata.Name); err != nil {
```

Comment:
If a component, the same component, has both input and output binding support, this will create two connections to the same component. Do you think there'd be a way to use a single connection perhaps?

Reply:
Yes, that's totally possible and I've been considering this - it's on track. Actually if you look closely you will see that this would happen not only for bindings but the same component can implement bindings, lock, statestore, pubsub and all building blocks at once, this is a matter of implementing all proto services that we have exposed,

However we decided to measure (and let this "improvement" for later) if having multiples connections vs single one is a problem considering the fact that we're using unix sockets which by the way is a very lightweight IPC protocol, it's always a tradeoff between throughput and compute resources,

\- connections / - resources usages / - throughput
\+ connections / + resource usages / + throughput

For bindings specifically, I like the idea of having a connection just for input binding since it's a continuous read stream.

wdyt @ItalyPaleAle ?

Reply:
I don't think it's likely that the same component will implement more than 1 component type (e.g. a binding + a state store), so I am not concerned with that.

If you are using UDS perhaps this is not as much of a concern. It would be more if you were using HTTP/2 as transport IMHO, as each connection uses another port and multiple connections don't allow the use of HTTP/2 multiplexing. But for UDS perhaps this is ok?

Reply:
> do you still think it's OK to keep both connections for now?

Given that you are using UDS rather than HTTP/2, perhaps it's ok ad interim. But if you think this is a small change and want to make it right away.... :)

Reply:
thanks!